tests/conftest.py

```diff
@@ -35,3 +36,46 @@ def clear_plugin_instances():
 
     request.addfinalizer(remove_log_files)
     request.addfinalizer(clear_plugin_instances)
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "requires_storage(required_space_gb=1, path='/'): Skip the test if insufficient disk space.",
+    )
+
+
+def check_storage(required_space_gb=1, path="/"):
+    """
+    Check the available disk space.
+
+    Args:
+        required_space_gb (float): Minimum required free space in GB.
+        path (str): Filesystem path to check.
+
+    Returns:
+        bool: True if there is enough free space, False otherwise.
+    """
+    total, used, free = shutil.disk_usage(path)
+    free_gb = free / (2**30)  # Convert bytes to gigabytes
+
+    return free_gb >= required_space_gb
+
+
+def pytest_runtest_setup(item):
+    """
+    Called before each test is run. Performs a storage check if a specific marker is present.
+    """
+    marker = item.get_closest_marker("requires_storage")
+    if marker:
+        required_space_gb = marker.kwargs.get("required_space_gb", 1)  # Default is 1GB
+        path = marker.kwargs.get("path", "/")  # Default is the root directory
+
+        if not check_storage(required_space_gb, path):
+            pytest.skip(
+                f"❌ Skipping test. Not enough free space ({required_space_gb} GB) at '{path}'."
+            )
+        else:
+            total, used, free = shutil.disk_usage(path)
+            free_gb = free / (2**30)  # Convert bytes to gigabytes
+            print(f"✅ Sufficient free space ({free_gb:.2f} GB) confirmed.")
```

Comment:
love it, thank you

we'll work on reducing this one day - would love to see any research/results you have on shrinking dep footprint, from other dev